app/services/GetCalculationDetailsService.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import connectors.httpParsers.CalculationDetailsHttpParser.CalculationDetailResponse
+import connectors.{CalculationDetailsConnector, GetCalculationListConnector}
+import uk.gov.hmrc.http.HeaderCarrier
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class GetCalculationDetailsService @Inject()(calculationDetailsConnector: CalculationDetailsConnector,
+                                             listCalculationDetailsConnector: GetCalculationListConnector) (implicit ec: ExecutionContext) {
+
+  def getCalculationDetails(nino: String, taxYear: String, OptionalTaxYear: Boolean)(implicit hc: HeaderCarrier):Future[CalculationDetailResponse] = {
```

Comment:
as mentioned on a previous PR if tax year can be optional the taxYear for listCalculationDetailsConnector.calcList method should be an Option[String] so you don't need the OptionalTaxYear boolean.